lti_consumer/lti_xblock.py

```diff
@@ -1120,6 +1153,10 @@ def lti_1p3_launch_callback(self, request, suffix=''):  # pylint: disable=unused
         Returns:
             webob.response: HTML LTI launch form or error page if misconfigured
         """
+        # TODO: I need to use the LTIConfiguration model for the version value here, but the consumer model
```

Comment:
Any ideas for this?

Reply:
I think it's ok to have a version attribute on the consumer class - using the name is just a brittle horrible version of doing that anyway.

You could have some sort of lookup table that maintains a mapping of the class itself to the version but again that's just a fancy version of having each class have to provide a version attribute. If in future we make LtiConsumerMagic1p3 that inherits from LtiConsumer1p3 it is a good feature that it would inherit version = lti_1p3 

If we make LtiConsumer1p3p5 we could override the version.

Or conversely you could make the consumer have a supports_version(string) function that lets you know that the lti1p3 supports 1.3 but not 1.3.5 and lti1p3p5 supports both, but that is maybe more fancy than we need now

Reply:
something we could look at here. You can get the consumer class through the LTIConfiguration model. At first glance this actually seems like a preferable way to do it since you get the lti config in scope for the version and the helper _get_lti_consumer just calls that anyway in a roundabout manner 🤔

Reply:
Thanks for the thoughts! Using `block.lti_version` means I don't have to deal with this issue!